specification/signalr/resource-manager/Microsoft.SignalR/preview/2018-03-01-preview/signalr.json

```diff
@@ -0,0 +1,1034 @@
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SignalR/SignalR/{resourceName}": {
+            "get": {
+                "tags": [
+                    "SignalR"
+                ],
+                "description": "Get the SignalR service and its properties.",
+                "operationId": "SignalR_Get",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceNameParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Success. The response describe the corresponding SingalR service.",
+                        "schema": {
+                            "$ref": "#/definitions/SignalRResourceResponse"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "Get": {
+                        "$ref": "./examples/Get.json"
+                    }
+                }
+            },
+            "put": {
+                "tags": [
+                    "SignalR"
+                ],
+                "description": "Create a new SignalR service and update an exiting SignalR service.",
+                "operationId": "SignalR_Create",
+                "parameters": [
+                    {
+                        "name": "createSignalRResourceRequest",
+                        "in": "body",
+                        "description": "parameters for the create or update operation",
+                        "required": false,
+                        "schema": {
+                            "$ref": "#/definitions/CreateSignalRResourceRequest"
+                        }
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceNameParameter"
+                    }
+                ],
+                "responses": {
+                    "201": {
+                        "description": "Created. The response describes the new service and contains a Location header to query the operation result.",
+                        "schema": {
+                            "$ref": "#/definitions/SignalRResourceResponse"
```

Comment:
The Model for the request and the response are different. Is it possible to make these the same? That'd enable us to round-trip an object (ie, GET, modify and PUT it back)

Reply:
@fearthecowboy should we use the same Definition for both request and response? or use different Definitions with the same properties?

Reply:
The same definition. That way the generated SDKs will get to use the same object.